common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -393,6 +393,20 @@ public void applicationRemoved(String appId, boolean cleanupLocalDirs) {
     }
   }
 
+  @Override
+  public void removeShuffleMerge(String appId, int shuffleId) {
```

Comment:
When adding support for `shuffleMergeId`, follow the same pattern as `finalizeShuffleMerge` - there are a few corner cases here, and that method handles them.